openmetadata-ui/src/main/resources/ui/src/components/common/CustomPropertyTable/PropertyValue.tsx

```diff
@@ -166,6 +296,342 @@ export const PropertyValue: FC<Props> = ({
+                      field: t('label.sql-uppercase-query'),
+                    }),
+                  },
+                ]}
+                style={commonStyle}
+                trigger="onChange">
+                <SchemaEditor
+                  className="custom-query-editor query-editor-h-200 custom-code-mirror-theme"
+                  mode={{ name: CSMode.SQL }}
+                  options={{
+                    readOnly: false,
+                  }}
+                />
+              </Form.Item>
+            </Form>
+          </InlineEdit>
+        );
+      }
+
+      case 'entityReference':
+      case 'entityReferenceList': {
+        const mode =
+          propertyType.name === 'entityReferenceList' ? 'multiple' : undefined;
+
+        const index = (property.customPropertyConfig?.config as string[]) ?? [];
+
+        let initialOptions: DataAssetOption[] = [];
+        let initialValue: string[] | string | undefined;
+
+        if (!isUndefined(value)) {
+          if (isArray(value)) {
+            initialOptions = value.map((item: EntityReference) => {
+              return {
+                displayName: getEntityName(item),
+                reference: item,
+                label: getEntityName(item),
+                value: item?.fullyQualifiedName ?? '',
+              };
+            });
+
+            initialValue = value.map(
+              (item: EntityReference) => item?.fullyQualifiedName ?? ''
+            );
+          } else {
+            initialOptions = [
+              {
+                displayName: getEntityName(value),
+                reference: value,
+                label: getEntityName(value),
+                value: value?.fullyQualifiedName ?? '',
+              },
+            ];
+
+            initialValue = value?.fullyQualifiedName ?? '';
+          }
+        }
+
+        const initialValues = {
+          entityReference: initialValue,
+        };
+
+        return (
+          <InlineEdit
+            isLoading={isLoading}
+            saveButtonProps={{
+              disabled: isLoading,
+              htmlType: 'submit',
+              form: 'entity-reference-form',
+            }}
+            onCancel={onHideInput}
+            onSave={noop}>
+            <Form
+              id="entity-reference-form"
+              initialValues={initialValues}
+              layout="vertical"
+              validateMessages={VALIDATION_MESSAGES}
+              onFinish={(values: {
+                entityReference: DataAssetOption | DataAssetOption[];
+              }) => {
+                if (isArray(values.entityReference)) {
+                  onInputSave(
+                    values.entityReference.map((item) => item.reference)
+                  );
+                } else {
+                  onInputSave(values.entityReference.reference);
+                }
+              }}>
+              <Form.Item
+                name="entityReference"
```

Comment:
Can you check if this three element can be someway optimised to reuse in each case?